src/panels/config/script/ha-script-picker.ts

```diff
@@ -1067,12 +1067,44 @@ ${rejected
     if (!entry) {
       return;
     }
-    await triggerScript(this.hass, entry.unique_id);
-    showToast(this, {
-      message: this.hass.localize("ui.notification_toast.triggered", {
-        name: computeStateName(script),
-      }),
-    });
+    try {
+      const config = await fetchScriptFileConfig(this.hass, entry.unique_id);
```

Comment:
I think the field definitions are already available locally at `hass.services.script[entry.unique_id].fields`? 

So you may not need to re-query it from backend.

Reply:
Let me check. The duplicate function was calling the backend just for the title so I followed a similar approach.

I will check. If this works, it is of course better

Reply:
Thanks for the right point. I actually found `hasScriptFields` function that can be reused and the changes are much simpler now. Please check again. Thanks.